### ISSUE TITLE
Build failure: python311Packages.qtile-extras on `nixpkgs/nixos-23.11`

### ISSUE THREAD

**Issue by juliendiot42** [NixOS/nixpkgs#271610]
Hello,

I wanted to switch to NixOS 23.11, but this build failure prevented it. I found a workaround (cf *Additional context*), but I am not sure it is very clean (it does not follow what is documented on [search.nixos.org](https://search.nixos.org/options?channel=23.11&show=services.xserver.windowManager.qtile.extraPackages&from=0&size=50&sort=relevance&type=packages&query=services.xserver.windowManager.qtile.extraPackages) and I am quite new to NixOs).

Note: Finally I will not use `qtile-extra` but since I couldn't found help about this problem I wanted to create this issue, it may help other people in the future.

Best regards,
JD

### Steps To Reproduce

Add `qtile-extra` to qtile as explained in the option's documentation: (cf. [services.xserver.windowManager.qtile.extraPackages](https://search.nixos.org/options?channel=23.11&show=services.xserver.windowManager.qtile.extraPackages&from=0&size=50&sort=relevance&type=packages&query=services.xserver.windowManager.qtile.extraPackages)) with `nixpkgs` targeting `nixos-23.11`.

In `configuration.nix`:
```
services.xserver.windowManager.qtile = {
      enable = true;
      backend = "x11";
      extraPackages = python3Packages: with python3Packages; [
        qtile-extras
      ];
};
```

In `flake.nix`:
```
nixpkgs.url = "nixpkgs/nixos-23.11";
```

`flake.lock`:

```
    "nixpkgs": {
      "locked": {
        "lastModified": 1660551188,
        "narHash": "sha256-a1LARMMYQ8DPx1BgoI/UN4bXe12hhZkCNqdxNi6uS0g=",
        "owner": "nixos",
        "repo": "nixpkgs",
        "rev": "441dc5d512153039f19ef198e662e4f3dbb9fd65",
        "type": "github"
      },
      "original": {
        "owner": "nixos",
        "repo": "nixpkgs",
        "type": "github"
      }
    },
```

### Build log

```
error: builder for '/nix/store/5v8k9bis2nqkwnskmb8j52pjjba37ccn-python3.11-qtile-extras-0.23.0.drv' failed with exit code 2;
       last 10 log lines:
       > test/widget/test_image.py:134
       >   /build/source/test/widget/test_image.py:134: PytestUnknownMarkWarning: Unknown pytest.mark.flaky - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
       >     @pytest.mark.flaky(reruns=5)
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > ERROR test/widget/test_strava.py - AttributeError: type object 'Activity' has no attribute 'RUN'
       > !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
       > ================= 125 deselected, 5 warnings, 1 error in 1.93s =================
       > /nix/store/wr08yanv2bjrphhi5aai12hf2qz5kvic-stdenv-linux/setup: line 1559: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/5v8k9bis2nqkwnskmb8j52pjjba37ccn-python3.11-qtile-extras-0.23.0.drv'.
```

### Additional context

I tryed to work around this problem and found a solution:

It works when doing in `configuration.nix`:

```
let
  qtile-extras-bis = pkgs.python310Packages.qtile-extras;
in
...
services.xserver.windowManager.qtile = {
      enable = true;
      backend = "x11";
      extraPackages = python3Packages: with python3Packages; [
        qtile-extras-bis
      ];
};
```

### Notify maintainers

@arjan-s 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.63, NixOS, 23.05 (Stoat), 23.05.20231124.1216a5b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.6`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### Priorities

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by TomaSajt** [comment]
From what I could tell from git bisecting, this broke in commit `8dfad603247387df1df4826b8bea58efc5d012d8` (this was a merge from the staging branch)
I'll do some more research on what exactly broke it.

**Comment by FliegendeWurst** [comment]
Fixed in #282869
